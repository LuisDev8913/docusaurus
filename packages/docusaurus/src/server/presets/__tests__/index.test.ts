/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';

import loadPresets from '../index';
import type {LoadContext} from '@docusaurus/types';

describe('loadPresets', () => {
  it('no presets', async () => {
    const context = {
      siteConfigPath: __dirname,
      siteConfig: {
        presets: [],
      },
    } as LoadContext;
    const presets = await loadPresets(context);
    expect(presets).toMatchInlineSnapshot(`
      Object {
        "plugins": Array [],
        "themes": Array [],
      }
    `);
  });

  it('string form', async () => {
    const context = {
      siteConfigPath: __dirname,
      siteConfig: {
        presets: [path.join(__dirname, '__fixtures__/preset-bar.js')],
      },
    } as LoadContext;
    const presets = await loadPresets(context);
    expect(presets).toMatchSnapshot();
  });

  it('string form composite', async () => {
    const context = {
      siteConfigPath: __dirname,
      siteConfig: {
        presets: [
          path.join(__dirname, '__fixtures__/preset-bar.js'),
          path.join(__dirname, '__fixtures__/preset-foo.js'),
        ],
      },
    } as LoadContext;
    const presets = await loadPresets(context);
    expect(presets).toMatchSnapshot();
  });

  it('array form', async () => {
    const context = {
      siteConfigPath: __dirname,
      siteConfig: {
        presets: [[path.join(__dirname, '__fixtures__/preset-bar.js')]],
      },
    } as Partial<LoadContext>;
    const presets = await loadPresets(context);
    expect(presets).toMatchSnapshot();
  });

  it('array form with options', async () => {
    const context = {
      siteConfigPath: __dirname,
      siteConfig: {
        presets: [
          [
            path.join(__dirname, '__fixtures__/preset-bar.js'),
            {docs: {path: '../'}},
          ],
        ],
      },
    } as Partial<LoadContext>;
    const presets = await loadPresets(context);
    expect(presets).toMatchSnapshot();
  });

  it('array form composite', async () => {
    const context = {
      siteConfigPath: __dirname,
      siteConfig: {
        presets: [
          [
            path.join(__dirname, '__fixtures__/preset-bar.js'),
            {docs: {path: '../'}},
          ],
          [
            path.join(__dirname, '__fixtures__/preset-foo.js'),
            {pages: {path: '../'}},
          ],
        ],
      },
    } as Partial<LoadContext>;
    const presets = await loadPresets(context);
    expect(presets).toMatchSnapshot();
  });

  it('mixed form', async () => {
    const context = {
      siteConfigPath: __dirname,
      siteConfig: {
        presets: [
          [
            path.join(__dirname, '__fixtures__/preset-bar.js'),
            {docs: {path: '../'}},
          ],
          path.join(__dirname, '__fixtures__/preset-foo.js'),
        ],
      },
    } as LoadContext;
    const presets = await loadPresets(context);
    expect(presets).toMatchSnapshot();
  });

  it('mixed form with themes', async () => {
    const context = {
      siteConfigPath: __dirname,
      siteConfig: {
        presets: [
          [
            path.join(__dirname, '__fixtures__/preset-bar.js'),
            {docs: {path: '../'}},
          ],
          path.join(__dirname, '__fixtures__/preset-foo.js'),
          path.join(__dirname, '__fixtures__/preset-qux.js'),
        ],
      },
    } as LoadContext;
    const presets = await loadPresets(context);
    expect(presets).toMatchSnapshot();
  });
});
