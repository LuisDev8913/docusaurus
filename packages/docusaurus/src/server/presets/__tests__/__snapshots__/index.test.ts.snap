// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadPresets array form 1`] = `
Object {
  "plugins": Array [
    Array [
      "@docusaurus/plugin-content-docs",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-content-blog",
      undefined,
    ],
  ],
  "themes": Array [],
}
`;

exports[`loadPresets array form composite 1`] = `
Object {
  "plugins": Array [
    Array [
      "@docusaurus/plugin-content-docs",
      Object {
        "path": "../",
      },
    ],
    Array [
      "@docusaurus/plugin-content-blog",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-content-pages",
      Object {
        "path": "../",
      },
    ],
    Array [
      "@docusaurus/plugin-sitemap",
      undefined,
    ],
  ],
  "themes": Array [],
}
`;

exports[`loadPresets array form with options 1`] = `
Object {
  "plugins": Array [
    Array [
      "@docusaurus/plugin-content-docs",
      Object {
        "path": "../",
      },
    ],
    Array [
      "@docusaurus/plugin-content-blog",
      undefined,
    ],
  ],
  "themes": Array [],
}
`;

exports[`loadPresets mixed form 1`] = `
Object {
  "plugins": Array [
    Array [
      "@docusaurus/plugin-content-docs",
      Object {
        "path": "../",
      },
    ],
    Array [
      "@docusaurus/plugin-content-blog",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-content-pages",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-sitemap",
      undefined,
    ],
  ],
  "themes": Array [],
}
`;

exports[`loadPresets mixed form with themes 1`] = `
Object {
  "plugins": Array [
    Array [
      "@docusaurus/plugin-content-docs",
      Object {
        "path": "../",
      },
    ],
    Array [
      "@docusaurus/plugin-content-blog",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-content-pages",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-sitemap",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-test",
      undefined,
    ],
  ],
  "themes": Array [
    Array [
      "@docusaurus/theme-classic",
      undefined,
    ],
  ],
}
`;

exports[`loadPresets string form 1`] = `
Object {
  "plugins": Array [
    Array [
      "@docusaurus/plugin-content-docs",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-content-blog",
      undefined,
    ],
  ],
  "themes": Array [],
}
`;

exports[`loadPresets string form composite 1`] = `
Object {
  "plugins": Array [
    Array [
      "@docusaurus/plugin-content-docs",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-content-blog",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-content-pages",
      undefined,
    ],
    Array [
      "@docusaurus/plugin-sitemap",
      undefined,
    ],
  ],
  "themes": Array [],
}
`;
